sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuAggregateExec.scala

```diff
@@ -280,15 +277,23 @@ object AggregateUtils extends Logging {
 
       val newBuckets = batchesByBucket.flatMap(bucket => {
         if (needRepartitionAgain(bucket)) {
-          val nextLayerBuckets =
-            ArrayBuffer.fill(hashBucketNum)(new AutoClosableArrayBuffer[SpillableColumnarBatch]())
-          // Recursively merge and repartition the over sized bucket
-          repartitionHappened =
-            iterateAndRepartition(
-              new CloseableBufferedIterator(bucket.iterator), metrics, targetMergeBatchSize,
-              helper, hashKeys, hashBucketNum, hashSeed + 7,
-              nextLayerBuckets) || repartitionHappened
-          nextLayerBuckets
+          if (hashSeed + 7 > 200) {
+            log.warn("Too many times of repartition, may hit a bug? Size for each batch in " +
+              "current bucket: " + bucket.map(_.sizeInBytes).mkString(", ") + " rows: " +
+              bucket.map(_.numRows()).mkString(", ") + " targetMergeBatchSize: "
+              + targetMergeBatchSize)
+            ArrayBuffer.apply(bucket)
```

Comment:
nit: could use the syntactic sugar here and elsewhere:

```suggestion
            ArrayBuffer(bucket)
```

Reply:
done

Reply:
But I still see the ArrayBuffer.apply(bucket)???

https://github.com/NVIDIA/spark-rapids/blob/738c8e38fc23c1634667443864b80f085f2737ac/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuAggregateExec.scala#L285

Did you forget to check in your last changes before you merged this? Or is there a follow on issue that you plan on doing.